### PR TITLE
Added the main body of code for Pachinko

### PR DESCRIPTION
**Pull request by Kortigus**

### CODE REVIEW

File: Pachinko/Pachinko.c [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+		printf(" %s\n", "Try again");
+		*credits -= *bet;
+	}
+}
+
+/* Drives the program */
+int main(void)
+{
+	int slot[3];
+	int credits = 1000;
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: Size_t instead of int

File: Pachinko/Pachinko.c [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+	int bet = 0;
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: Same as above

File: Pachinko/Pachinko.c [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+
+	srand((unsigned)time(NULL));
+
+	while (1)
+	{
+		if (credits < 100)
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: Use MINIMUM instead of 100

File: Pachinko/Pachinko.c [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+	char line[LINESIZE];
+	*bet = 0;
+	while (1)
+	{
+		printf("How much would you like to bet?");
+		if (!fgets(line, LINESIZE, stdin))
+		{
+			clearerr(stdin);
+			printf("Error reading line");
+			return 1;
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: If there's an error, and this returns zero, does a player keep playing?

File: Pachinko/Pachinko.c [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+		}
+		if (sscanf_s(line, "%d", bet) == 1 && *bet >= 100)
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: Use MINIMUM instead of 100

File: Pachinko/Pachinko.c [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+	*number = rand() % 4;
+}
+
+/* Spins the slots */
+void playSlots(int slot[], size_t n)
+{
+	size_t i;
+	for (i = 0; i < 3; i++)
+	{
+		pickRandom(&slot[i]);
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: You don't need to make this into a function.

File: Pachinko/Pachinko.c [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+/* Chooses a random number between 0 and 4 inclusively */
+void pickRandom(int* number)
+{
+	*number = rand() % 4;
+}
+
+/* Spins the slots */
+void playSlots(int slot[], size_t n)
+{
+	size_t i;
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c:11]: Erase this and use n

File: Pachinko/Pachinko.c [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+/* Checks if the slots match and pays the player accordingly */
+void payOut(int slot[], int* credits, int* bet)
+{
+	size_t i;
+	int* p = &slot[0];
+	for (i = 0; i < 3; i++)
+	{
+		switch (slot[i])
+		{
+		case 0: printf("%s", "@");
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: Use the following format
switch(slot[i])
{
case 0:
{
    printf("%s", "@");
    break;
}
case 1:
...
}

File: Pachinko/Pachinko.c [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+			    break;
+		case 1: printf("%s", "#");
+				break;
+		case 2: printf("%s", "$");
+				break;
+		case 3: printf("%s", "%");
+				break;
+		case 4: printf("%s", "7");
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: You can't have 4. (rand() % 4 generates 0 to 3)

File: Pachinko/Pachinko.c [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+		}
+		printf("%s\n", "The minimum to bet is 100 Credits, please pick a larger number");
+	}
+}
+
+/* Checks if the slots match and pays the player accordingly */
+void payOut(int slot[], int* credits, int* bet)
+{
+	size_t i;
+	int* p = &slot[0];
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: Why do you need this?
Just use slot

**Review by ExtremeSkepticist — CHANGES_REQUESTED** [review]

File: Pachinko/Pachinko.c (hunk 7 above)
Comment (Kortigus) [reply Pachinko/Pachinko.c:11]: should i erase i, or use it to iterate, and then replace the > 3 with > n

Comment (ExtremeSkepticist) [reply Pachinko/Pachinko.c:11]: No thj sis a mistake. I was gonna say use n instead of 3.

File: Pachinko/Pachinko.c [hunk 11]
```diff
@@ -5,68 +5,86 @@
...
+	for (i = 0; i < n; i++)
 	{
-		pickRandom(&slot[i]);
+		slot[i] = rand() % 5;
 	}
 }
 
 /* Gets the amount the player wants to bet and stores it into bet */
-int insertCoin(int* bet)
+insertCoin(size_t* bet, size_t* credits)
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c]: Where's the return type?

File: Pachinko/Pachinko.c [hunk 12]
```diff
@@ -5,68 +5,86 @@
...
 	*bet = 0;
 	while (1)
 	{
-		printf("How much would you like to bet?");
+		printf("How much would you like to bet?\n");
 		if (!fgets(line, LINESIZE, stdin))
 		{
 			clearerr(stdin);
 			printf("Error reading line");
 			return 1;
```
Comment (ExtremeSkepticist) [inline Pachinko/Pachinko.c:30]: If there's an error, you shouldn't let the roulette spin and let a user start over. That's what I meant

**Review by ExtremeSkepticist — CHANGES_REQUESTED** [review]

**Pull request merged**